server/src/main/java/org/apache/iotdb/db/utils/QueryUtils.java

```diff
@@ -112,36 +122,16 @@ public static void filterQueryDataSource(QueryDataSource queryDataSource,
     unseqResources.removeIf(fileFilter::fileNotSatisfy);
   }
 
-  public static void constructPathAndDataTypes(List<PartialPath> paths, List<TSDataType> dataTypes,
-      List<ShowTimeSeriesResult> timeseriesList) {
-    paths.add(new PartialPath(COLUMN_TIMESERIES, false));
-    dataTypes.add(TSDataType.TEXT);
-    paths.add(new PartialPath(COLUMN_TIMESERIES_ALIAS, false));
-    dataTypes.add(TSDataType.TEXT);
-    paths.add(new PartialPath(COLUMN_STORAGE_GROUP, false));
-    dataTypes.add(TSDataType.TEXT);
-    paths.add(new PartialPath(COLUMN_TIMESERIES_DATATYPE, false));
-    dataTypes.add(TSDataType.TEXT);
-    paths.add(new PartialPath(COLUMN_TIMESERIES_ENCODING, false));
-    dataTypes.add(TSDataType.TEXT);
-    paths.add(new PartialPath(COLUMN_TIMESERIES_COMPRESSION, false));
-    dataTypes.add(TSDataType.TEXT);
-
-    Set<String> tagAndAttributeName = new TreeSet<>();
-    for (ShowTimeSeriesResult result : timeseriesList) {
-      tagAndAttributeName.addAll(result.getTagAndAttribute().keySet());
-    }
-    for (String key : tagAndAttributeName) {
-      paths.add(new PartialPath(key, false));
-      dataTypes.add(TSDataType.TEXT);
-    }
+  public static void constructPathAndDataTypes(List<PartialPath> paths, List<TSDataType> dataTypes) {
+    Collections.addAll(paths, resourcePaths);
+    Collections.addAll(dataTypes, resourceTypes);
   }
 
   public static QueryDataSet getQueryDataSet(List<ShowTimeSeriesResult> timeseriesList,
       ShowTimeSeriesPlan showTimeSeriesPlan, QueryContext context) {
     List<PartialPath> paths = new ArrayList<>();
     List<TSDataType> dataTypes = new ArrayList<>();
-    constructPathAndDataTypes(paths, dataTypes, timeseriesList);
+    constructPathAndDataTypes(paths, dataTypes);
```

Comment:
I think that `constructPath` and `updateRecord` behavior belongs to `Dataset`, how about move it to `ShowTimeseriesDataSet ` class?

and next line ,  the `showTimeSeriesPlan` already exists in dataset...


@qiaojialin  what do you think?

Reply:
It seems that the constructPathAndDataTypes is only used for ShowTimeseries, so moving it to ShowTimeseriesDataSet is ok.